src/Pact/Native.hs

```diff
@@ -518,6 +521,15 @@ filterDef = defNative "filter" filter'
   ["(filter (compose (length) (< 2)) [\"my\" \"dog\" \"has\" \"fleas\"])"]
   "Filter LIST by applying APP to each element. For each true result, the original value is kept."
 
+distinctDef :: NativeDef
+distinctDef = defGasRNative "distinct" distinct
+  (funType (TyList a) [("values", TyList a)] <>
+   funType (TyList (tTyObject (mkSchemaVar "o"))) [("fields", TyList tTyString), ("values", TyList (tTyObject (mkSchemaVar "o")))])
+  ["(distinct [3 3 1 1 2 2])", "(distinct ['age] [{'name: \"Lin\",'age: 30} {'name: \"Val\",'age: 25}])"]
+  $ T.intercalate " "
+  [ "Returns from a homogeneous list of VALUES, or objects using supplied FIELDS list, a list with duplicates removed."
```

Comment:
Why did we decide we need FIELDS? It's much simpler to not offer that.

Reply:
I was taking inspiration from sort and thought it might be useful. But I can understand why it might not be necessary.